### PR TITLE
[7.x] Add assertResource assertion for TestResponse

### PR DESCRIPTION
**Pull request by AidynMakhataev**
This PR adds a new assertion to the TestResponse class called assertResource. Here is an example:

```php
$item = factory(Item::class)->create();

$response = $this->get(route('items.show'));

// for single resource
$response->assertResource(new ItemResource($item));

// for collection
$response->assertResource(ItemResource::collection($items));

```

### CODE REVIEW

File: src/Illuminate/Testing/TestResponse.php [hunk 1]
```diff
@@ -1149,6 +1150,17 @@ public function assertSessionMissing($key)
         return $this;
     }
 
+    /**
+     * Assert that the response is a given resource.
+     *
+     * @param JsonResource $resource
```
Comment (GrahamCampbell) [inline src/Illuminate/Testing/TestResponse.php:1156]: Please use the fully qualified class name.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Testing/TestResponse.php (hunk 1 above)
Comment (AidynMakhataev) [reply src/Illuminate/Testing/TestResponse.php]: fixed

File: src/Illuminate/Testing/TestResponse.php [hunk 2]
```diff
@@ -1153,10 +1152,10 @@ public function assertSessionMissing($key)
     /**
      * Assert that the response is a given resource.
      *
-     * @param JsonResource $resource
+     * @param \Illuminate\Http\Resources\Json\JsonResource $resource
      * @return $this
      */
-    public function assertResource(JsonResource $resource)
+    public function assertResource(\Illuminate\Http\Resources\Json\JsonResource $resource)
```
Comment (GrahamCampbell) [inline src/Illuminate/Testing/TestResponse.php:1158]: Please don't use fully qualified names in the actual type here. Only in the phpdoc, like the rest of the code.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by owenconti** [comment]
> This PR adds a new assertion to the TestResponse class called assertResource. Here is an example:

This is a good idea and will be useful for tests. It’d also be useful if you could specify an optional key to look for a nested resource.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**